receiver/awsxrayreceiver/internal/proxy/server.go

```diff
@@ -0,0 +1,165 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package proxy provides an http server to act as a signing proxy for SDKs calling AWS X-Ray APIs
+package proxy
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	v4 "github.com/aws/aws-sdk-go/aws/signer/v4"
+	"go.uber.org/zap"
+)
+
+const (
+	service    = "xray"
+	connHeader = "Connection"
+)
+
+// Server represents HTTP server.
+type Server interface {
+	ListenAndServe() error
+	Close() error
+}
+
+// NewServer returns a local TCP server that proxies requests to AWS
+// backend using the given credentials.
+func NewServer(cfg *Config, logger *zap.Logger) (Server, error) {
+	_, err := net.ResolveTCPAddr("tcp", cfg.Endpoint)
+	if err != nil {
+		return nil, err
+	}
+	if cfg.ProxyAddress != "" {
+		logger.Debug("Using remote proxy", zap.String("address", cfg.ProxyAddress))
+	}
+
+	awsCfg, sess, err := getAWSConfigSession(cfg, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	awsEndPoint, err := getServiceEndpoint(awsCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	logger.Info("HTTP proxy server using remote X-Ray endpoint", zap.String("value", awsEndPoint))
```

Comment:
This log statement seems out of place, especially since there's a chance that the endpoint parses incorrectly and server doesn't start. Can we replace with just "x-ray HTTP proxy server started" somewhere after it's started?